pkg/logs/scheduler/scheduler.go

```diff
@@ -247,6 +257,16 @@ func (s *Scheduler) toService(config integration.Config) (*service.Service, erro
 	return service.NewService(provider, identifier, s.getCreationTime(config)), nil
 }
 
+// extractMainSourceAndService extracts the source & service attached to the container config
+func (s *Scheduler) extractMainSourceAndService(configs []*logsConfig.LogsConfig) (string, string) {
```

Comment:
@prognant It's unclear to me when this logic is useful, and why we need to extract a source and service from the entire slice of `LogsConfig`s. Also I'm not sure what `container config` refers to (does that refer to a docker label/pod annotation, or something different?).

Could you clarify this and add a unit test that covers a scenario where this logic is useful? (and maybe refer to this unit test from code comments, so that there's a clear example of when it's useful)

Reply:
It makes sense to detail a bit this piece of code, the main use case (for now) is to handle annotation looking like :

```
[{"source":"mySource","service":"myService"}, {"type":"file","path":"/tmp/share/test.log"}]
```

And copy source/service to the file config (else it will get an empty service/source) as the foreseen behaviour was to get the same metadata attached to the lines collected from the file and the ones collected from the container.

Reply:
Interesting example: is this an annotation that is "created" by the Agent's autodiscovery logic, or does it really reflect an annotation that a user would set?

Because, to me, it is expected that your example for the `logs` annotation:

```json
[{"source":"mySource","service":"myService"}, {"type":"file","path":"/tmp/share/test.log"}]